library/L1_Peripheral/lpc17xx/system_controller.hpp

```diff
@@ -117,273 +53,563 @@ class SystemController final : public sjsu::SystemController
+    };
+
+    /// Namespace containing the bit masks for the PLL1 Configuration register
+    /// (PLL1CFG).
+    struct ConfigurationRegister  // NOLINT
+    {
+      /// PLL1 multiplier bit mask
+      static constexpr bit::Mask kMultiplierMask =
+          bit::CreateMaskFromRange(0, 4);
+      /// PLL1 pre-divider bit mask
+      static constexpr bit::Mask kDividerMask = bit::CreateMaskFromRange(5, 6);
+    };
+
+    /// Namespace containing the bit masks for the PLL1 Status register
+    /// (PLL1STAT).
+    struct StatusRegister  // NOLINT
+    {
+      /// PLL1 multiplier bit mask
+      static constexpr bit::Mask kMultiplierMask =
+          bit::CreateMaskFromRange(0, 4);
+      /// PLL1 pre-divider bit mask
+      static constexpr bit::Mask kDividerMask = bit::CreateMaskFromRange(5, 6);
+      /// PLL1 enable status bit mask.
+      static constexpr bit::Mask kEnableMask = bit::CreateMaskFromRange(8);
+      /// PLL1 connection status bit mask.
+      static constexpr bit::Mask kConnectMask = bit::CreateMaskFromRange(9);
+      /// PLL1 lock status bit
+      static constexpr bit::Mask kLockStatusMask = bit::CreateMaskFromRange(10);
+    };
   };
 
-  /// Fixed IRC frequency
-  static constexpr units::frequency::hertz_t kDefaultIRCFrequency = 4_MHz;
+  /// Namespace containing the common bit masks for the PLL Control registers
+  /// (PLL0CON and PLL1CON).
+  struct PllControlRegister  // NOLINT
+  {
+    /// PLL enable bit mask.
+    static constexpr bit::Mask kEnableMask = bit::CreateMaskFromRange(0);
+    /// PLL connect/disconnect bit mask.
+    static constexpr bit::Mask kConnectMask = bit::CreateMaskFromRange(1);
+  };
 
-  /// Required frequency for the USB clock
-  static constexpr units::frequency::hertz_t kUsbClockFrequency = 48_MHz;
+  /// Namespace containing the bit masks for the CPU Clock Configuration
+  /// register (CCLKCFG).
+  struct CpuClockRegister  // NOLINT
+  {
+    /// The 8-bit CPU clock divider bitmask.
+    static constexpr bit::Mask kDividerMask = bit::CreateMaskFromRange(0, 7);
+  };
 
-  /// Fixed RTC frequency
+  /// Namespace containing the bit masks for the USB Clock Configuration
+  /// register (USBCLKCFG).
+  struct UsbClockRegister  // NOLINT
+  {
+    /// The 4-bit USB clock divider bitmask.
+    static constexpr bit::Mask kDividerMask = bit::CreateMaskFromRange(0, 3);
+  };
+
+  /// @see 4.1 Summary of clocking and power control functions
+  ///      https://www.ti.com/lit/ug/slau356i/slau356i.pdf#page=29
+  struct ClockConfiguration_t
+  {
+    /// Configurations for the main oscillator. This oscillator can be used as a
+    /// clock source to drive the system clock or PLL1.
+    struct
+    {
+      /// Frequency of the main oscillator ranging between 1 MHz to 25 MHz.
+      ///
+      /// @note This value will vary and should be set based on the external
+      ///       oscillator that is being used.
+      units::frequency::hertz_t frequency = 0_MHz;
+    } main_oscillator;
```

Comment:
```suggestion
    } main_oscillator = {};
```
To make sure that this member is default intialized.